pkg/cluster/space/rook.go

```diff
@@ -0,0 +1,154 @@
+package clusterspace
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"code.cloudfoundry.org/bytefmt"
+	rookcli "github.com/rook/rook/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/replicatedhq/kurl/pkg/k8sutil"
+)
+
+const (
+	namespace = "rook-ceph"
+)
+
+// RookChecker checks if we have enough disk space to migrate volumes to rook.
+type RookChecker struct {
+	kcli  kubernetes.Interface
+	rcli  rookcli.Interface
+	cfg   *rest.Config
+	log   *log.Logger
+	srcSC string
+	dstSC string
+}
+
+// freeSpace attempts to get the ceph free space. returns the number of available bytes.
+func (r *RookChecker) freeSpace(ctx context.Context) (int64, error) {
```

Comment:
wdyt about use the prefix get to make clear what the func does? 
```suggestion
func (r *RookChecker) getFreSpaceWith(ctx context.Context) (int64, error) {
```

Reply:
I don't like to use prefix `get` in these scenarios. This seems more like a java than a go thing. I haven't understood the `With` as well, can you elaborate ?

Reply:
I tend to prefer the idiomatic Go, like [here](https://go.dev/doc/effective_go#Getters). That being said, the function has been renamed as suggested.

Reply:
Hi @ricardomaraschini,

> I tend to prefer the idiomatic Go, like [here](https://go.dev/doc/effective_go#Getters). That being said, the function has been renamed as suggested. 

But in this case we have struct.Spec(). Yes, in this case we do need get/set. 
However, in the code here we have an func. So, ihmo both are not the same scenario.

Reply:
Hi @ricardomaraschini,

> I tend to prefer the idiomatic Go, like [here](https://go.dev/doc/effective_go#Getters). That being said, the function has been renamed as suggested.

Thank you for the reply, and allow me to clarify my thoughts, understanding and motivations on this one.

The suggestion here does not go against ` idiomatic Go` concept. See in the doc(link shared): 
`and **it's often appropriate to do so**, but it's neither idiomatic nor necessary to put Get into the getter's name`. 

Therefore, we do not need to implement getter/setter for all as that usually is done in JAVA, as you point out, for example. _(that does not see fit well with the stack either)_. Then, when it is implicit that it is a get, it is not required to use Get. Therefore, that also does not mean never using GET as a prefix. It shows means use when it is required only for readability.  

So, in this case,  we do not have `validator.Validator()` such as in the doc example, where we know what it is returning. We have validator.freeSpace() then: is that returning the free space amount OR doing an operation to ensure a freeSpace? That was my motivation for the suggestion.